src/main/java/com/statsig/androidsdk/Statsig.kt

```diff
@@ -7,447 +7,444 @@ import android.content.SharedPreferences
+        event.user = user
+
+        if (!options.disableCurrentActivityLogging) {
+            val className = lifecycleListener.currentActivity?.javaClass?.simpleName
+            if (className != null) {
+                event.statsigMetadata = mapOf("currentPage" to className)
             }
+        }
+        statsigScope.launch {
             logger.log(event)
         }
+    }
 
-        /**
-         * Log an event to Statsig for the current user
-         * @param eventName the name of the event to track
-         * @param value an optional value assocaited with the event
-         * @param metadata an optional map of metadata associated with the event
-         * @throws IllegalStateException if the SDK has not been initialized
-         */
-        @JvmStatic
-        fun logEvent(eventName: String, value: String, metadata: Map<String, String>? = null) {
-            enforceInitialized("logEvent")
-            if (this.state == null) {
-                return
-            }
-            var event = LogEvent(eventName)
-            event.value = value
-            event.metadata = metadata
-            event.user = this.user
+    /**
+     * Log an event to Statsig for the current user
+     * @param eventName the name of the event to track
+     * @param value an optional value assocaited with the event
+     * @param metadata an optional map of metadata associated with the event
+     * @throws IllegalStateException if the SDK has not been initialized
+     */
+    @JvmOverloads
+    @JvmStatic
+    fun logEvent(eventName: String, value: String, metadata: Map<String, String>? = null) {
+        enforceInitialized("logEvent")
+        if (state == null) {
+            return
+        }
+        val event = LogEvent(eventName)
+        event.value = value
+        event.metadata = metadata
+        event.user = user
+        statsigScope.launch {
             logger.log(event)
         }
+    }
 
-        /**
-         * Log an event to Statsig for the current user
-         * @param eventName the name of the event to track
-         * @param metadata an optional map of metadata associated with the event
-         * @throws IllegalStateException if the SDK has not been initialized
-         */
-        @JvmStatic
-        fun logEvent(eventName: String, metadata: Map<String, String>) {
-            enforceInitialized("logEvent")
-            if (this.state == null) {
-                return
-            }
-            var event = LogEvent(eventName)
-            event.value = null
-            event.metadata = metadata
-            event.user = this.user
+    /**
+     * Log an event to Statsig for the current user
+     * @param eventName the name of the event to track
+     * @param metadata an optional map of metadata associated with the event
+     * @throws IllegalStateException if the SDK has not been initialized
+     */
+    @JvmStatic
+    fun logEvent(eventName: String, metadata: Map<String, String>) {
```

Comment:
Looking at this again - I'm not sure we actually need this method?  Since the above jvm overloads method has an optional metadata parameter, we should already have this version of the function.  At the very least, this function could call the other logEvent function so we dont duplicate the implementation